.github/workflows/test-go.yml

```
@@ -241,7 +241,7 @@ jobs:
           # shellcheck disable=SC2086 # can't quote package list
 
           # On a release branch, add a flag to rerun failed tests
-          if [[ ${{ github.base_ref }} = release/* ]] ; then
+          if [[ ${{ github.ref_name }} == 'release/*' ]]; then
```

I think you need to wrap `${{ github.ref_name }}` in quotes and then "release/*" rather than 'release/*':
```suggestion
          if [[ "${{ github.ref_name }}" == "release/*" ]]; then
```